functions.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * Function to connect to the database
+ * @param string $DBname
+ * @return PDO
+ */
+function connectToDB(string $DBname): PDO
+{
+// 1. Database Connection arguments
+    $host = 'db';
+    $user = 'root';
+    $password = 'password';
+
+    $dsn = "mysql:host=$host;dbname=$DBname";
+
+    $options = [
+        PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC,
+        PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION
+    ];
+
+    try {
+        return new PDO($dsn, $user, $password, $options);
+    } catch (\PDOException $exception) {
+        throw new \PDOException($exception->getMessage(), (int)$exception->getCode());
+    }
+}
+
+/**
+ * Function to get data from database
+ * @param PDO $pdo
+ * @return array
+ */
+function getCollection($pdo): array
+{
+    //1. Prepare statement
+    $query = $pdo->prepare(
+        'SELECT 
+        `routes`.`name`,
+        `routes`.`distance`,
+        `routes`.`discipline`,
+        `routes`.`effort_level`,
+        `routes`.`adrenaline_rating`,
+        `routes`.`short_description`,
+        `locations`.`country_name`,
+        `locations`.`region`,
+        `images`.`url`,
+        `images`.`alt_text`
+        FROM `routes`
+        LEFT JOIN `locations` on `locations`.`location_id`=`routes`.`location_id`
+        RIGHT JOIN `images` on `images`.`route_id`=`routes`.`id`;'
+    );
+
+    //2. Execute query
+    $query->execute();
+    return $query->fetchAll();
+}
+
+
+/**
+ * Function to return a string representing HTML for each collection item card
+ * @param array $array
+ * @return string
+ */
+function displayCollection(array $array)
```

Comment:
Type hint the return.

Reply:
Added type hint return of string to this function